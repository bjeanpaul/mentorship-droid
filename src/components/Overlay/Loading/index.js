import React, { PropTypes } from 'react';
import { StyleSheet } from 'react-native';

import { BaseView, Text, Image, SpinningImage } from 'src/components';

import colors from 'src/constants/colors';
import IMAGE_DISMISS_ICON from 'app/assets/close-light.png';


const styles = StyleSheet.create({
  baseView: {
    padding: 24,
    backgroundColor: colors.OVERLAY_BG,
    alignItems: 'center',
    justifyContent: 'center',
  },
  dismissButton: {
    position: 'absolute',
    left: 16,
    top: 16,
  },
  title: {
    fontSize: 20,
    color: colors.OVERLAY_TEXT,
    paddingBottom: 80,
  },
});

const LoadingOverlay = ({
  title,
  dismiss,
}) => {
  let dismissButton;

  if (dismiss) {
    dismissButton = (
      <Image
        style={styles.dismissButton}
        source={IMAGE_DISMISS_ICON}
      />
    );
  }

  return (
    <BaseView
      statusBarBackgroundColor={colors.OVERLAY_STATUS_BG}
      style={styles.baseView}
    >
      {dismissButton}
      <SpinningImage />
      <Text style={styles.title}>{title}</Text>
    </BaseView>
  );
};
LoadingOverlay.propTypes = {
  dismiss: PropTypes.bool,
  title: PropTypes.string.isRequired,
};
export default LoadingOverlay;
