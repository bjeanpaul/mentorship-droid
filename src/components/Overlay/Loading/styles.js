import { StyleSheet } from 'react-native';
import colors from 'src/constants/colors';

export default StyleSheet.create({
  baseView: {
    padding: 24,
    backgroundColor: colors.OVERLAY_BG,
    alignItems: 'center',
    justifyContent: 'center',
  },
  dismissButton: {
    position: 'absolute',
    left: 16,
    top: 16,
  },
  title: {
    fontSize: 20,
    color: colors.OVERLAY_TEXT,
    paddingBottom: 80,
  },
});
