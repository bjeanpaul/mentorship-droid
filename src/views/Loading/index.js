import React from 'react';
import { OverlayLoading } from 'src/components';


const Loading = () => (
  <OverlayLoading title="Loading your information..." />
);


export default Loading;
