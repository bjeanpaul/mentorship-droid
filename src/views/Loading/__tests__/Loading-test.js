import React from 'react';
import Loading from 'src/views/Loading';


describe('Loading', () => {
  it('should render', () => {
    const el = render(<Loading />);
    expect(el).toMatchSnapshot();
  });
});
