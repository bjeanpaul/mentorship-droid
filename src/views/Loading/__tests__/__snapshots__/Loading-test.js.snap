exports[`Loading should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(246, 111, 63, 0.95)",
        "justifyContent": "center",
        "padding": 24
      }
    ]
  }>
  <Image
    source={1}
    style={
      Object {
        "height": 80,
        "transform": Array [
          Object {
            "rotate": "0deg"
          }
        ],
        "width": 80
      }
    } />
  <Text
    style={
      Array [
        Object {
          "color": "#003035",
          "fontFamily": "Brandon Text",
          "fontSize": 13,
          "textAlign": "center"
        },
        Object {
          "color": undefined,
          "fontSize": 20,
          "paddingBottom": 80
        }
      ]
    }>
    Loading your information...
  </Text>
</View>
`;
